src/keystore/InMemoryKeystore.ts

```diff
@@ -251,28 +254,139 @@ export default class InMemoryKeystore implements Keystore {
       const invitation = InvitationV1.createRandom(req.context)
       const created = nsToDate(req.createdNs)
       const recipient = toSignedPublicKeyBundle(req.recipient)
-      const sealed = await SealedInvitation.createV1({
-        sender: this.v2Keys,
+
+      return await this.makeInvite(
+        this.v2Keys,
         recipient,
         created,
-        invitation,
-      })
-      const topicData = {
-        invitation,
-        createdNs: req.createdNs,
-        peerAddress: await recipient.walletSignatureAddress(),
+        req.createdNs,
+        invitation
+      )
+    } catch (e) {
+      throw convertError(e as Error, ErrorCode.ERROR_CODE_INVALID_INPUT)
+    }
+  }
+
+  async createInvites(
+    req: keystore.CreateInvitesRequest
+  ): Promise<keystore.CreateInviteResponse[]> {
+    try {
+      if (!validateObject(req, ['recipients'], [])) {
+        throw new KeystoreError(
+          ErrorCode.ERROR_CODE_INVALID_INPUT,
+          'missing recipients'
+        )
       }
-      await this.inviteStore.add([topicData])
 
-      return keystore.CreateInviteResponse.fromPartial({
-        conversation: topicDataToConversationReference(topicData),
-        payload: sealed.toBytes(),
+      // Register an address on the network for the group
+      const sharedWallet = Wallet.createRandom()
+      const sharedClient = await Client.create(sharedWallet)
```

Comment:
I thought we were getting rid of the shared client with these changes?

Reply:
This is just to have an address for the group to use as a peer address. I'm creating a client here so that the contact bundle exists on the network.

Reply:
Are we sure that's necessary with the invite approach. Couldn't the invites just come from whoever invited you to the group chat. No new contact bundles, no mysterious addresses.

Reply:
I'm trying to think how that would work in the UI. With a dedicated address for each group, we could just show group chats as normal entries. It's also a way to verify that you're actually in the same group as someone else, like "Oh are you in 0x1234...1234?"

Reply:
It turns out I have a bug here and I actually am sending the invitation from the creator's address. I still think product-wise, it makes sense for groups to have their own addresses though.

Reply:
> With a dedicated address for each group, we could just show group chats as normal entries.

Why would we not be able to show group chats as normal entries if the invites came from the group creator (or inviter)?

> It's also a way to verify that you're actually in the same group as someone else, like "Oh are you in 0x1234...1234?"

Couldn't we use the topic for that? Or generate a random group ID and set that as metadata


Getting rid of the random wallet address feels like a big win for maintainability and simplicity IMO.

Reply:
> Getting rid of the random wallet address feels like a big win for maintainability and simplicity IMO.

Ok, I'm sold on this. I'll update to set a `groupID` in the metadata alongside the member list. We'll have to make sure clients know to display these differently in their conversation list.